### PR TITLE
closeAllL1Desc modifications + test (4-2-stable)

### PR DESCRIPTION
**Pull request by alanking**
Addresses https://github.com/irods/irods/issues/6622

Cherry-picked from https://github.com/irods/irods/pull/6676

Tests are running

### CODE REVIEW

File: scripts/irods/test/test_rulebase.py [hunk 1]
```diff
@@ -381,6 +382,63 @@ def test_create_close__issue_5018(self):
...
+
+                # Create an iRODS rule which opens a data object at logical_path, calls msiExecCmd, and then closes the data object.
+                rule_file = os.path.join(self.admin.local_session_dir, 'rule.r')
+                with open(rule_file, 'w') as f:
+                    f.write(rule_map[self.plugin_name].format(logical_path))
+
+                self.admin.assert_icommand(['irule', '-r', self.plugin_name + '-instance', '-F', rule_file])
+
+                # msiExecCmd should not execute any policy. Since the rule calls msiDataObjClose only once, there should be only
+                # AVU associated with the data object. If this were not the case, there would be two: once for the msiDataObjClose
```
Comment (korydraughn) [inline scripts/irods/test/test_rulebase.py:431]: ... there should be **only AVU** associated ...?

Comment (alanking) [reply scripts/irods/test/test_rulebase.py]: Done

**Comment by alanking** [comment]
Tests are passing

**Comment by korydraughn** [comment]
Are the tests still in progress?

**Comment by alanking** [comment]
The tests passed. Tense is hard :)

**Review by korydraughn — APPROVED** [review]
Squash and add the #'s

**Pull request merged**